### PR TITLE
Remove discovery service and IFMAP server

### PR DESCRIPTION
**Pull request by Doude**
From release R4.0, contrail replaced discovery service by static
configurations of entry points in each service configuration files.
IFMAP server was also removed from config nodes.
Also force to use python Thrift 0.9.3 version which is a pycassa
constraints. Pycassa don't work with Thrift 0.10.0 release.

### CODE REVIEW

File: devstack/plugin.sh [hunk 1]
```diff
@@ -223,15 +223,34 @@ elif [[ "$1" == "stack" && "$2" == "pre-install" ]]; then
...
         sudo -E scons $SCONS_ARGS
         cd $TOP_DIR
 
         # As contrail's python packages requirements aren't installed
         # automatically, we have to manage their installation.
-        pip_install -r $CONTRAIL_PLUGIN_DIR/files/requirements.txt
-
+        if _vercmp $CONTRAIL_BRANCH ">=" R4.0; then
+            sed '/^discoveryclient$/d' $CONTRAIL_PLUGIN_DIR/files/requirements.txt > /tmp/contrail_requirements.txt
+            pip_install -r /tmp/contrail_requirements.txt
```
Comment (zioc) [inline devstack/plugin.sh]: What about removing discoveryclient from requirements.txt, and installing it manually if _vercmp $CONTRAIL_BRANCH "<" R4.0 ?

File: devstack/plugin.sh [hunk 2]
```diff
@@ -223,15 +223,34 @@ elif [[ "$1" == "stack" && "$2" == "pre-install" ]]; then
...
+        else
+            pip_install -r $CONTRAIL_PLUGIN_DIR/files/requirements.txt
+        fi
+        # Force to use the version 0.9.3 of the Thrift python library as it is a requirements for Pycassa library.
+        # Recent OpenStack release require at least Thrift 0.10.0.
+        sudo pip install -U thrift==0.9.3
```
Comment (zioc) [inline devstack/plugin.sh]: This was (maybe partially) adressed by commit [0989d1eb76e9e151932e45d4fdc913991bcb8e3](https://github.com/zioc/contrail-devstack-plugin/commit/e0989d1eb76e9e151932e45d4fdc913991bcb8e3)

Wouldn't it be better to put thrift==0.9.3 in requirements.txt ?

File: devstack/plugin.sh (hunk 1 above)
Comment (Doude) [inline devstack/plugin.sh]: Yes probably better

File: devstack/plugin.sh (hunk 2 above)
Comment (Doude) [inline devstack/plugin.sh]: I tried that but recent OpenStack release force to upgrade to 0.10.0 even if we require the 0.9.3. OpenStack force that with [requirements tool](https://github.com/openstack/requirements/blob/master/upper-constraints.txt#L112).
Do you know if we can add some constraints from the devstack plugin to the OpenStack requirements?

**Comment by zioc** [comment]
Thanks!

**Pull request merged**